modules/lfs/content_store.go

```diff
@@ -60,6 +60,12 @@ func (s *ContentStore) Put(pointer Pointer, r io.Reader) error {
 		return err
 	}
 
+	// check again whether there is any error during the Save operation
+	// because some errors might be ignored by the Reader's caller
+	if wrappedRd.lastError != nil && !errors.Is(wrappedRd.lastError, io.EOF) {
+		return wrappedRd.lastError
```

Comment:
If `err == nil` and `wrappedRd.lastError != nil` the file was uploaded and needs to get removed. Currently only the error is returned but the file is still there.